packages/phone-number-privacy/src/index.ts

```diff
@@ -52,3 +55,42 @@ function hasValidPhoneNumberParam(requestBody: any): boolean {
 function hasValidQueryPhoneNumberParam(requestBody: any): boolean {
   return requestBody.queryPhoneNumber
 }
+
+// TODO (amyslawson) consider pagination or streaming of contacts?
+export const getContactMatches = functions.https.onRequest(async (request, response) => {
+  // TODO (amyslawson) refactor this internal logic and input validation to its own file
+  // when adding error handling
+  try {
+    if (!isValidGetContactMatchesInput(request.body)) {
+      response.status(400).send('Invalid input parameters')
+      return
+    }
+    authenticateUser()
+    const matchedContacts: ContactMatch[] = (
+      await getNumberPairContacts(request.body.userPhoneNumber, request.body.contactPhoneNumbers)
+    ).map((numberPair) => ({ phoneNumber: numberPair, salt: computeBLSSalt(numberPair) }))
+    await setNumberPairContacts(request.body.userPhoneNumber, request.body.contactPhoneNumbers)
+    // TODO (amyslawson) return salts with contact
+    response.json({ success: true, matchedContacts })
+  } catch (e) {
+    console.log('Failed to getContactMatches', e)
+    response.status(500).send(e)
```

Comment:
Don't send e, and ditto got getSalt. It leaks details of our internal implementation, which is sometimes abused by attackers. Less applicable since this is open source but still good general policy

Reply:
kk i will update this in the next pr that focuses on the error handling